tools/cellxgene_census_builder/src/cellxgene_census_builder/build_soma/experiment_builder.py

```diff
@@ -475,72 +480,101 @@ def _accumulate_all_X_layers(
         if ad.n_obs == 0:
             continue
 
-        # follow CELLxGENE 3.0 schema conventions for raw/X aliasing when only raw counts exist
-        raw_X, raw_var = (ad.X, ad.var) if ad.raw is None else (ad.raw.X, ad.raw.var)
-
-        if not is_nonnegative_integral(raw_X):
-            logging.error(f"{dataset.dataset_id} contains non-integer or negative valued data")
-
-        if isinstance(raw_X, np.ndarray):
-            raw_X = sparse.csr_matrix(raw_X)
-
-        raw_X.eliminate_zeros()
-
         # save X['raw']
         layer_name = "raw"
         logging.info(
             f"{eb.name}: saving X layer '{layer_name}' for dataset '{dataset.dataset_id}' "
             f"({progress[0]} of {progress[1]})"
         )
-        local_var_joinids = raw_var.join(eb.global_var_joinids).soma_joinid.to_numpy()
+        local_var_joinids = ad.var.join(eb.global_var_joinids).soma_joinid.to_numpy()
         assert (local_var_joinids >= 0).all(), f"Illegal join id, {dataset.dataset_id}"
 
-        for n, X in enumerate(array_chunker(raw_X), start=1):
-            logging.debug(f"{eb.name}/{layer_name}: X chunk {n} {dataset.dataset_id}")
+        # accumulators
+        obs_stats: pd.DataFrame = pd.DataFrame()
+        var_stats: pd.DataFrame = pd.DataFrame()
+
+        for idx in range(0, ad.n_obs, ACCUM_X_STRIDE):
+            logging.debug(f"{eb.name}/{layer_name}: X chunk {idx//ACCUM_X_STRIDE + 1} {dataset.dataset_id}")
+
+            X = ad[idx : idx + ACCUM_X_STRIDE].X
+            if isinstance(X, np.ndarray):
+                X = sparse.csr_matrix(X)
+
+            if not is_nonnegative_integral(X):
+                logging.error(f"{dataset.dataset_id} contains non-integer or negative valued data")
```

Comment:
The current code is identical to previous behavior.  That said, good question - if this fails, it indicates a dataset that has violated the schema. We don't have a strict policy about how these are handled, but for many errors it will cause a hard failure. So I think the right approach is to simply change this to an assert.